### DISCUSSION TITLE
Layout Header/Footer/Sidebar Slot Extension

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#8941] [Feature Requests]
We have layouts for browsing and visualizing collection items in different ways, but it would be nice to have open-ended options for injecting more information/tools there. Some initial ideas:

* **Chart** — A configurable chart that updates based on the current filtered data. Perhaps reusing Insight panels?
* **Video Player** — A sticky header for playing videos would turn Directus into YouTube.
* **Audio Player** — A sticky footer would turn Directus into an audio player
* **Aggregate Data Footer** — Could display different aggregate functions (SUM, MAX, MIN, etc) for each column. Not sure if this makes sense for layouts other than Table... perhaps we look to scope these to Layouts like interfaces to types? 🤔 
* **Activity / Time Series** — Similar to chart, this could show activity based on a date/time field.
* **Item Preview / Edit** — Could open a header (or sidebar?) for viewing the selected item inline. Instead of opening the Item Page/Drawer, this would keep the Layout listing available while also enabling viewing/editing the item.
* **Filter Header** — Could add a persistent inline filter section that is always onscreen (instead of within search)
* **Map** — An interesting idea, you could have a Table/Card/Timeline/etc layout, but with a Map header/footer for showing the current results (or selected items)
* **Breadcrumb** — Not sure if this fits, but when browsing relational data it could show "where" you are in the tree, and a button to go "Back"
* **Toolbar** — Configurable actions for the current collection and/or selected items. For example, if you had an e-com collection in Table view, you could have buttons to _Refund_ or _Approve_ the selected items.

**Comment by danieljwestman** [comment]
Nice ideas!

"Item Preview" (a customizable view of an item, not a form) 👌🏻

I wrote something I still stand for ref #4930, "Item Presentation View - A new extension type"

**Comment by addisonElliott** [comment]
This would be a big plus to have! To add onto the ideas, there's a react-admin tool used to build custom admin dashboards. I was originally planning on using that and building a custom dashboard, but stumbling upon Directus had made me reconsider using Directus instead.

React-admin demo here: https://marmelab.com/react-admin-demo/#/products/126

Here's a view ideas with visuals from react-admin.

**Tabbed view for edit**

![image](https://user-images.githubusercontent.com/1342819/138555386-441b14b8-1b81-432d-aa90-48fad34ee01c.png)

**Custom context items on selection**

![image](https://user-images.githubusercontent.com/1342819/138555371-b4b529e1-7f63-45b0-9d4d-b2121a1112a6.png)

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by rijkvanzanten** [comment]
An additional slot could be "table row end" in the tabular layout or other layout specific places (as per https://github.com/directus/directus/discussions/18678)
